src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -288,7 +288,7 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
 
       const returnFiber = workInProgress.return;
 
-      if (returnFiber) {
+      if (returnFiber && !next) {
```

Comment:
Actually, it's because the new test uses classes which have "update effects" scheduled on them to invoke the life-cycles. Those effects gets scheduled twice.